src/cascadia/TerminalSettingsModel/ActionMapSerialization.cpp

```diff
@@ -49,6 +56,30 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         return warnings;
     }
 
+    Json::Value ActionMap::ToJson() const
+    {
+        Json::Value actionList{ Json::ValueType::arrayValue };
+        for (const auto& [_, cmd] : _ActionMap)
+        {
+            const auto cmdImpl{ winrt::get_self<implementation::Command>(cmd) };
+            const auto& cmdJsonArray{ cmdImpl->ToJson() };
+            for (const auto& cmdJson : cmdJsonArray)

```

Comment:
So wait, if I had 
```jsonc  
{
    "icon": "whatever icon",
    "name": "whatever name...",
    "commands": [
        {
            "command": { "action": "splitPane", "profile": "a name 1", "split": "auto" }
        },
        {
            "command": { "action": "splitPane", "profile": "a name 1", "split": "vertical" }
        },
        {
            "command": { "action": "splitPane", "profile": "a name 1", "split": "horizontal" }
        }
    ]
}
```

(note that this is _not_ an iterable command)

originally, is that going to:
* first get serialized as-is, by the `if (_nestedCommand)` branch in `Command::ToJson`, THEN
* get serialized _again_ by the `for (const auto& [_, cmd] : _NestedCommands)` loop below as 
```json
{
    "command": { "action": "splitPane", "profile": "a name 1", "split": "auto" }
},
{
    "command": { "action": "splitPane", "profile": "a name 2", "split": "vertical" }
},
{
    "command": { "action": "splitPane", "profile": "a name 3", "split": "horizontal" }
}
```

For a total of 4 top-level commands: one with three nested beneath it, and 3 more top-level commands?

Or is the original top-level command (`whatever name...`) only in `_NestedCommands`, so it's not an issue?

Reply:
Nah, we only do it once. Here's what the trace looks like:
1. `ActionMap::AddAction()`
   - if the `Command` is the root of a nested command, add the root to `_NestedCommands`. We DO NOT add the sub commands to `_ActionMap`.
   - otherwise, add it to `_ActionMap`
2. `ActionMap::ToJson()`
   - serialize all of the `_ActionMap` commands
      3. `Command::ToJson` --> <more complicated, out of scope of this question>
   - serialize all of the `_NestedCommands` commands
      3. `Command::ToJson` --> just spit out the `Command::_originalJson`

So since we don't actually record the sub-commands in `ActionMap`, we don't ever write them out other than via `Command::_originalJson`.

NOTE: _theoretically_, we could just always return `Command::_originalJson`. But I'm sensing that we shouldn't depend on that.